polyfactory/factories/base.py

```diff
@@ -660,7 +660,7 @@ def process_kwargs(cls, **kwargs: Any) -> dict[str, Any]:
         for field_meta in cls.get_model_fields():
             field_build_parameters = cls.extract_field_build_parameters(field_meta=field_meta, build_args=kwargs)
             if cls.should_set_field_value(field_meta, **kwargs):
-                if hasattr(cls, field_meta.name) and field_meta.name in cls.__dict__:
+                if hasattr(cls, field_meta.name) and not hasattr(BaseFactory, field_meta.name):
```

Comment:
how would this change work with custom base factories?

Reply:
I don't think this is a good solution. 

Rather i would traverse the MRO and filter attributes belonging to base factories out.

Reply:
Not sure I am following. The whole point of this PR is to inherit the attributes of the classes along the MRO (except for `BaseFactory` itself), not filter them out (which is what currently happens).

Reply:
But should you inherit the attributes of base factories?  This will interfere with declaration of Fields

Reply:
Well basically any attribute or method name that could collide. You should consider the possibility of users shooting themselves in to foot unintentionally - i.e. "foolproof" the code. 

If a user defines a baseclass with a no underscored method or attribute that is not committed here, it might conflict with regular factory fields.

Reply:
First of all, the attribute must have the same name in the respective `__model__` for this to happen. We're iterating over `cls.get_model_fields()`, not the whole factory `__dict__`. A base factory can freely introduce new non-field methods/attributes (underscored or not) if their name is not in the `__model__`.

Now If the name _does_ exist in the model, one can already shoot themselves in the foot if a (non-base) factory attribute is a callable that takes parameters, e.g. any regular `f(self, ...)` method or a `@property`. To make it foolproof, fields would need to be more restricted, for example by requiring them to extend a new `polyfactory.Field` class (which would be a backwards incompatible change and orthogonal to this PR anyway).

Reply:
Ok, makes sense